tests/Doctrine/Tests/DBAL/Functional/Schema/SqliteSchemaManagerTest.php

```diff
@@ -257,4 +257,42 @@ public function testOnlyOwnCommentIsParsed(): void
             ->getColumn('col1')
             ->getComment());
     }
+
+    public function testUnnamedForeignKeyConstraintHandling(): void
+    {
+        $sql = <<<SQL
+create table "bug4243_users" (
+"id" integer not null primary key autoincrement
+);
+create table "bug4243_posts"
+("id" integer not null primary key autoincrement,
+"user_id" integer not null,
+"body" varchar not null,
+"created_at" datetime null,
+"updated_at" datetime null,
+foreign key("user_id") references "bug4243_users"("id"));
+SQL;
+
+        $this->connection->exec($sql);
+
+        $sm          = $this->connection->getSchemaManager();
+        $onlineTable = $sm->listTableDetails('bug4243_posts');
+
+        $offlineTable = new Table('bug4243_posts');
+        $offlineTable->addColumn('id', 'integer');
+
+        $comparator = new Schema\Comparator();
+        $diff       = $comparator->diffTable($offlineTable, $onlineTable);
+
+        $sqls = $this->connection->getDatabasePlatform()->getAlterTableSQL($diff);
+
+        foreach ($sqls as $sql) {
+            $this->connection->exec($sql);
+        }
+
+        $onlineTableAfter = $sm->listTableDetails('bug4243_posts');
+        $diff             = $comparator->diffTable($onlineTable, $onlineTableAfter);
+
+        $this->assertFalse($diff);
```

Comment:
I'm unfamiliar with how this works… is `$onlineTable` supposed to be mutated at some point to be the same as `$onlineTableAfter`, i.e. a table with just an `id` column and nothing more?

Reply:
no, $onlineTable is the one create from the first exec(). the offline table has only the id, so the iteration over $sqls essentially recreates the existing table, because sqlite does not really do an ALTER.

Then after the applied SQLs, the onlineTableAfter should be exactly the same as the onlineTable.

Reply:
Ok so in `$comparator->diffTable($offlineTable, $onlineTable);$sqls = $this->connection->getDatabasePlatform()->getAlterTableSQL($diff);`, `$offlineTable` is the starting point, and `$onlineTable` is the goal, is that it? So we are pretending `$offlineTable` is what we have in the DB, when in fact we have more columns, and it works anyway because we start by dropping the table?

Reply:
@greg0ire exactly.